server.js

```diff
@@ -2,37 +2,105 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import data from './data/qa.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/questions"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+// Reusable variable for error messages
+const ERR_NO_QUESTIONS = 'Sorry, could not find this question.'
+
+// Question Model
+const Question = mongoose.model('Question', {
+  id: {
+    type: Number
+  },
+  question: {
+    type: String
+  },
+  likes: {
+    type: Number
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now()
+  }
+})
+ 
+//Seeding database
+
+if (process.env.RESET_DB) {
+  console.log('Resetting Database!')
+  const seedDatabase = async () => {
+    await Question.deleteMany({})
+    data.forEach((question) => {
+      new Question(question).save()
+    })
+  }
+  seedDatabase()
+}
 
+
+
+const port = process.env.PORT || 9000
+const app = express()
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// Adding middleware for targeting server error for every request
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable'})
+  }
+})  
+
+// My endpoints 
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(`<h2>Hello!</h2><h4>Possible endpoints for this API. </h4><ul><li>'/questions' shows all questions in the database</li><li>'/questions?query={'search word'}' gives the user the possibility to enter a search word to look for a question in a specific topic.</li><li>'/questions/:id' finds a specific question by id number</li><li>'/popular' returns the most popular questions based on number of likes</li><li>'/nolikes' return the questions that have not yet recieved any likes.</li></ul>`)
```

Comment:
Nice with documentation! ⭐ Take a look at this npm package for listing endpoints: [https://www.npmjs.com/package/express-list-endpoints](https://www.npmjs.com/package/express-list-endpoints)